src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+import tagComponent from '../../../utils/helpers/tags';
+import DatePicker from './date-picker.component';
+import InputDecoratorBridge from '../input-decorator-bridge';
+import StyledDateInput from './date.style';
+import Textbox from '../textbox';
+
+const isoDateFormat = 'YYYY-MM-DD';
+const today = DateHelper.todayFormatted(isoDateFormat);
+
+/**
+ * A Date widget.
+ *
+ * == How to use a Date in a component:
+ *
+ * In your file
+ *
+ *   import Date from 'carbon-react/lib/components/Date';
+ *
+ * To render the Date:
+ *
+ *   <Date name="myDate" />
+ *
+ * @class Date
+ * @constructor
+ * @decorators {Input,InputIcon,InputLabel,InputValidation}
+ */
+class Date extends React.Component {
+  static propTypes = {
+    ...Textbox.propTypes,
+    /** Automatically focus on component mount */
+    autoFocus: PropTypes.bool,
+    /** Used to provide additional validations on composed components */
+    internalValidations: PropTypes.array,
+    /** Minimum possible date YYYY-MM-DD */
+    minDate: PropTypes.string,
+    /** Maximum possible date YYYY-MM-DD */
+    maxDate: PropTypes.string,
+    /** Specify a callback triggered on blur */
+    onBlur: PropTypes.func,
+    /** Specify a callback triggered on change */
+    onChange: PropTypes.func,
+    /** Specify a callback triggered on focus */
+    onFocus: PropTypes.func,
+    /** The current date */
+    value: PropTypes.string
+  };
+
+  static defaultProps = {
+    value: today,
+    internalValidations: [new DateValidator()]
+  };
+
+  // Stores the document - allows us to override it different contexts, such as when running tests.
+  _document = document;
+
+  bridge = React.createRef() // this is a reference to the input decorator bridge component
+
+  blurBlocked = false; // stops the blur callback from triggering (closing the list) when we don't want it to
+
+  state = {
+    isDatePickerOpen: false,
+    selectedDate: null,
+    visibleValue: this.formatVisibleValue(this.props.value)
+  };
+
+  componentDidMount() {
+    if (this.props.autoFocus) {
+      this.blockFocus = true;
+      this.input.focus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.blurBlocked && this.hasDatePickerValueChanged(prevProps)) {
+      this.unblockBlur();
+      this.handleBlur(); // TODO validate
+    }
+  }
+
+  hasDatePickerValueChanged = (prevProps) => {
+    return this.props.value && prevProps.value !== this.props.value;
+  };
+
+  assignInput = (input) => {
+    this.input = input.current;
+  };
+
+  blockBlur() {
+    this.blurBlocked = true;
+    this.bridge.current.blockBlur = true; // this is to support the legacy behaviour in the bridge
+  }
+
+  unblockBlur() {
+    this.blurBlocked = false;
+    this.bridge.current.blockBlur = false; // this is to support the legacy behaviour in the bridge
+  }
+
+  handleBlur = (ev) => {
+    if (this.blurBlocked) return;
+    if (this.props.onBlur) this.props.onBlur(ev);
+  }
+
+  handleFocus = (ev) => {
+    if (this.blockFocus) {
```

Comment:
I'm finding this a little hard to decode. Perhaps assign it in the same way that you are with `blurBlocked`, with a comment? 

I'm not clear from the name whether or not this would be a bool or a func., currently is reads like an instruction.

Reply:
you're right, it's a leftover from the previous Decorator based implementation, I'll change it.